src/icastats_shared.c

```diff
@@ -277,14 +277,14 @@ void stats_increment(stats_fields_t field, int hardware, int direction)
 
 	if(direction == ENCRYPT)
 		if (hardware == ALGO_HW)
-			atomic_add((int *)&stats[field].enc.hw, 1);
+			atomic_add((uint64_t *)&stats[field].enc.hw, 1);
```

Comment:
Why do we need a cast here? Are the fields not defined as uint64_t already? 
Neding a cast most always points to a coding problem....

Reply:
ok, I removed the cast.